lib/waffle/storage/google/cloud_storage.ex

```diff
@@ -71,9 +69,9 @@ defmodule Waffle.Storage.Google.CloudStorage do
   Constructs a new connection object with scoped authentication. If no scope is
   provided, the `devstorage.full_control` scope is used as a default.
   """
-  @spec conn(String.t) :: Tesla.Env.client
-  def conn(scope \\ @full_control_scope) do
-    {:ok, token} = Goth.Token.for_scope(scope)
+  @spec conn(Types.definition) :: Tesla.Env.client
+  def conn(definition) do
+    {:ok, token} = Goth.fetch(definition.goth_module())
```

Comment:
on eweay to do this without having to pass the module is to use GenServer functions to retrieve the config and get a  token

```
config_process = GenServer.whereis(Goth.Config)
%{default: config} = :sys.get_state(pid)
{:ok, token} = Goth.Token.fetch(config)
```

Since the tool relies heavily on Goth anyways this should get you what we need.

Reply:
Goth.Config and Goth.Token.fetch are deprecated.